contracts/AssetPool.sol

```diff
@@ -253,13 +260,75 @@ contract AssetPool is Ownable {
         underwriterToken.burn(covAmount);
     }
 
+    /// @notice Withdraws collateral tokens to a new Asset Pool which previously
+    ///         was approved by the governance. Old underwriter tokens are burned
+    ///         in favor of new minted tokens that proof share of
+    ///         collateral tokens in a new Asset Pool.
+    function withdrawToNewAssetPool() external {
+        /* solhint-disable not-rely-on-time */
+        require(
+            address(newAssetPool) != address(0),
+            "New asset pool must be assigned"
+        );
+
+        uint256 covUnderwriterBalance = underwriterToken.balanceOf(msg.sender);
+
+        require(
+            covUnderwriterBalance > 0,
+            "Underwriter's token balance must be greater than 0"
+        );
+
+        uint256 covSupply = underwriterToken.totalSupply();
+
+        // slither-disable-next-line reentrancy-events
+        rewardsPool.withdraw();
+
+        uint256 collateralBalance = collateralToken.balanceOf(address(this));
+
+        uint256 amountToWithdraw =
+            covUnderwriterBalance.mul(collateralBalance).div(covSupply);
+
+        collateralToken.safeTransfer(address(newAssetPool), amountToWithdraw);
+        emit WithdrawalToNewAssetPoolCompleted(
+            msg.sender,
+            amountToWithdraw,
+            block.timestamp
+        );
+
+        // mappings cleanup since all the underwriter's funds are transferred to
+        // a new asset pool
+        delete withdrawalInitiatedTimestamp[msg.sender];
+        delete pendingWithdrawal[msg.sender];
+
+        underwriterToken.burn(covUnderwriterBalance);
+        newUnderwriterToken.mint(msg.sender, covUnderwriterBalance);
```

Comment:
Our token implementation has only one owner that is allowed to mint it. We can rework the token to allow for multiple minters but from a design perspective, it feels better if there was only one minter for a token.

Reply:
Okay, I take that the problem will pop up with the old AssetPool being an owner of `newUnderwriterToken` and a new AssetPool won't be able to mint `newUnderwriterTokens`? If yes, wdyt about transferring an ownership to a new AssetPool right after we mint new tokens, ie. `newUnderwriterToken.transferOwnership(address(newAssetPool))`?

Reply:
yea, that comment was not relevant anymore.